servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -672,7 +680,16 @@ public void channelInactive(ChannelHandlerContext ctx) {
         }
 
         private void doChannelActive(ChannelHandlerContext ctx) {
+            if (active) {
+                return;
+            }
+            active = true;
+
             if (waitForSslHandshake) {
+                if (observer != null) {
+                    // Notify ConnectionObserverHandler that we are ready to receive SecurityHandshakeObserver instance
+                    ctx.pipeline().fireUserEventTriggered(WaitingForHandshakeCompletionEvent.INSTANCE);
```

Comment:
A negative of sending an event is that there is no acknowledgment of receipt which makes debugging harder.
An alternative is to follow what we do for `DeferSslHandler` => lookup the handler and register interest for receiving the observer. The negative is more public API (internal so arguably low) but gives a more determinsitic behavior.

Reply:
Good suggestion! Everything is pkg-private without public exposure 🎉